src/libasr/asdl_cpp.py

```diff
@@ -658,23 +658,24 @@ def visitModule(self, mod):
         self.emit("public:")
         self.emit(  "std::string s, indtd;", 1)
         self.emit(  "bool use_colors;", 1)
-        self.emit(  "bool start_line = true;", 1)
         self.emit(  'bool last, attached;', 1)
-        self.emit(  "int indent_level = 0, indent_spaces = 2, lvl = 0;", 1)
+        self.emit(  "std::vector<int> indent_length;", 1)
         self.emit("public:")
         self.emit(  "TreeBaseVisitor() : use_colors(false), last(true), attached(false) { s.reserve(100000); }", 1)
         self.emit(  "void inc_indent() {", 1)
-        self.emit(      "indent_level++;", 2)
+        self.emit(      "indent_length.push_back(2);", 2)
         self.emit(      'indtd += "  ";', 2)
         self.emit(  "}", 1)
         self.emit(  "void inc_lindent() {", 1)
-        self.emit(      "indent_level++;", 2)
-        self.emit(      'indtd += "| ";', 2)
+        self.emit(      "indent_length.push_back(4);", 2)
+        self.emit(      'indtd += "│ ";', 2)
```

Comment:
Previously the problem with this was, the `│` character occupies 3 bytes but we tried to remove only the 2 bytes (using substr), so the `?` symbol appeared in the tree.

Reply:
See: https://stackoverflow.com/a/33349765/15913193
`│` -> 0xE2